### PR TITLE
Feature/update brew version

### PR DESCRIPTION
**Pull request by vimanyu**
Fix for errors observed while installing packages with homebrew. Until Github runners are updated to use latest hombrew versions, manually updating homebrew in Github workflows.

https://github.com/Homebrew/brew/pull/11070
https://github.com/Homebrew/discussions/discussions/691

### CODE REVIEW

**Review by jonsimantov — COMMENTED** [review]
Do the Integration tests and CPP packaging workflows need this as well?

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -78,6 +78,14 @@ jobs:
           path: /tmp/android-ndk-r16b
           key: android-ndk-${{ matrix.os }}-r16b
 
+      - name: Update homebrew (avoid bintray errors)
+        if: startsWith(matrix.os, 'macos')
```
Comment (jonsimantov) [inline .github/workflows/android.yml:82]: any particular reason this uses startsWith(matrix.os) and the below uses runner.os ==? Can you unify them?

**Review by jonsimantov — CHANGES_REQUESTED** [review]

**Comment by vimanyu** [comment]
Thanks for pointing it out. Yes, they needed this update too. 
I have scanned through the usages of install_prereqs scripts in our workflows and added this step prior.
Running integration_tests workflow just to confirm.

File: .github/workflows/android.yml (hunk 1 above)
Comment (vimanyu) [reply .github/workflows/android.yml:82]: So I was trying to keep it consistent with any other macos checks in those blocks. 
For a couple of the blocks, I found the comparison used runner.os and for others, we used macos.

Comment (vimanyu) [reply .github/workflows/android.yml:82]: For example, cpp-packaging uses `runner.os` comparison,
https://github.com/firebase/firebase-cpp-sdk/commit/c87c8a4abf71d3c31f8f29fd9b4ca04d68b289f2

integration_tests.yml uses "matrix.os"
https://github.com/firebase/firebase-cpp-sdk/commit/ce9c9fa52ae3f96f43971b2040b545c439efbc00

**Review by jonsimantov — APPROVED** [review]

**Pull request merged**